### PR TITLE
asserts: deserialize grouping only once in Pool.AddBatch if needed

### PR DESCRIPTION
**Pull request by pedronis**
this covers an XXX

### CODE REVIEW

**Comment by pedronis** [comment]
this is based on #9400

**Review by bboozzoo — COMMENTED** [review]
The last patch LGTM

File: asserts/pool.go [hunk 1]
```diff
@@ -521,11 +525,29 @@ func (p *Pool) Add(a Assertion, grouping Grouping) (ok bool, err error) {
...
+	deser := func(_ string) (*internal.Grouping, error) {
+		if cachedGrouping != nil {
+			// do a copy as addToGrouping and resolveWith
+			// might add to their input
+			g := cachedGrouping.Copy()
+			return &g, nil
+		}
+		var err error
+		cachedGrouping, err = p.groupings.Deserialize(string(grouping))
+		return cachedGrouping, err
```
Comment (stolowski) [inline asserts/pool.go:545]: Nitpick, these 3 lines could be reduced to `return p.groupings.Deserialize(string(grouping))`?

**Review by zyga — APPROVED** [review]
LGTM, interesting ride into assertions code.

**Review by stolowski — APPROVED** [review]
LGTM, one nitpick. Thank you

File: asserts/pool.go (hunk 1 above)
Comment (pedronis) [reply asserts/pool.go:545]: no, is setting cachedGrouping from line 535, otherwise it would stay nil and there would be no caching

Comment (stolowski) [reply asserts/pool.go:545]: Ah indeed, I missed that, thanks.

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**